Code/Tools/AssetProcessor/native/AssetManager/PathDependencyManager.cpp

```diff
@@ -29,7 +35,93 @@ namespace AssetProcessor
+    {
+        if (m_queuedForResolve.empty())
+        {
+            return;
+        }
+
+        auto queuedForResolve = m_queuedForResolve;
+        m_queuedForResolve.clear();
+
+        // Grab every product from the database and map to Source PK -> [products]
+        AZStd::unordered_map<AZ::s64, AZStd::vector<AzToolsFramework::AssetDatabase::ProductDatabaseEntry>> productMap;
+        m_stateData->QueryCombined([&productMap](const AzToolsFramework::AssetDatabase::CombinedDatabaseEntry& entry)
+        {
+            productMap[entry.m_sourcePK].push_back(entry);
+            return true;
+        });
+
+        // Build up a list of all the paths we need to search for: products + 2 variations of the source path
+        AZStd::vector<SearchEntry> searches;
+
+        for (const auto& entry : queuedForResolve)
+        {
+            // Search for each product
+            for (const auto& productEntry : productMap[entry.m_sourceID])
+            {
+                const AZStd::string& productName = productEntry.m_productName;
+
+                // strip path of the <platform>/
+                AZStd::string_view result = AssetUtilities::StripAssetPlatformNoCopy(productName);
+                searches.emplace_back(result, false, &entry, &productEntry);
+            }
+
+            // Search for the source path
+            AZStd::string sourceNameWithScanFolder =
+                ToScanFolderPrefixedPath(aznumeric_cast<int>(entry.m_scanFolderPK), entry.m_sourceName.c_str());
+            AZStd::string sanitizedSourceName = entry.m_sourceName;
+
+            SanitizeForDatabase(sourceNameWithScanFolder);
+            SanitizeForDatabase(sanitizedSourceName);
+
+            searches.emplace_back(sourceNameWithScanFolder, true, &entry, nullptr);
+            searches.emplace_back(sanitizedSourceName, true, &entry, nullptr);
+        }
+
+        AzToolsFramework::AssetDatabase::ProductDependencyDatabaseEntryContainer unresolvedDependencies;
+        m_stateData->GetUnresolvedProductDependencies(unresolvedDependencies);
+
+        AZStd::recursive_mutex mapMutex;
+        // Map of <Source PK => Map of <Matched SearchEntry => Product Dependency>>
+        AZStd::unordered_map<AZ::s64, AZStd::unordered_map<const SearchEntry*, AZStd::unordered_set<AzToolsFramework::AssetDatabase::ProductDependencyDatabaseEntry>>> map;
+
+        // For every search path we created, we're going to see if it matches up against any of the unresolved dependencies
+        AZ::parallel_for_each(
+            searches.begin(), searches.end(),
+            [&map, &mapMutex, &unresolvedDependencies](const SearchEntry& search)
+            {
+                AZStd::unordered_set<AzToolsFramework::AssetDatabase::ProductDependencyDatabaseEntry> matches;
+                for (const auto& entry: unresolvedDependencies)
+                {
+                    if (AZStd::wildcard_match(entry.m_unresolvedPath, search.m_path))
```

Comment:
FYI there is a [AZ::IO::PathView::Match](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/IO/Path/Path.h#L224-L243) that can match a path against a glob expression.